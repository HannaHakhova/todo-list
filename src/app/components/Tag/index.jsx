import React, {Component} from 'react';
import {Label, Icon} from 'semantic-ui-react';
import {string, func, shape, number} from 'prop-types';

class Tag extends Component {
    onDragStart = (e, v) => {
        const ev = e;
        ev.dataTransfer.dropEffect = 'move';
        ev.dataTransfer.setData('text/plain', v);
    }

    render() {
        return (
            <div>
                <Label as='a' tag color='teal' draggable='true' onDragStart={e => this.onDragStart(e, this.props.tag.id)}> {this.props.tag.title}
                    <Label.Detail>{this.props.tag.count}</Label.Detail>
                    <Icon name='delete' onClick={() => this.props.handleDelete(this.props.tag.id)} />
                </Label>
            </div>
        );
    }
}

Tag.propTypes = {
    tag: shape({
        id: string.isRequired,
        title: string.isRequired,
        count: number.isRequired
    }).isRequired,
    handleDelete: func.isRequired
};

export default Tag;