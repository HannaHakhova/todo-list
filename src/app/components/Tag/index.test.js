/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import {shallow} from 'enzyme';
import Tag from './index';

describe('Tag', () => {
    it('matches expected snapshot)', () => {
        expect(shallow(<Tag
            handleDelete={() => {}}
            tag={{id: '123456', title: 'test'}} />)).toMatchSnapshot();
    });

    it('calls handleDelete when it is clicked', () => {
        const handleDelete = jest.fn();
        const wrapper = shallow(<Tag
            handleDelete={handleDelete}
            tag={{id: '123456', title: 'test'}} />);

        wrapper.find('Icon').simulate('click');

        expect(handleDelete).toBeCalledWith('123456');
    });
});