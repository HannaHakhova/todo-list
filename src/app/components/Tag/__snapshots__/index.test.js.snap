// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag matches expected snapshot) 1`] = `
<div>
  <Label
    as="a"
    color="teal"
    draggable="true"
    onDragStart={[Function]}
    tag={true}
  >
     
    test
    <LabelDetail />
    <Icon
      as="i"
      name="delete"
      onClick={[Function]}
    />
  </Label>
</div>
`;
